docs/developer/extending/plugins.mdx

```diff
@@ -193,3 +195,134 @@ Plugin as a response needs to return the object of type inherited from `django.h
+    OpenIDConnectPlugin ->>+ OAuthProvider: fetch tokens
+    OAuthProvider -->>- OpenIDConnectPlugin: tokens
+    OpenIDConnectPlugin ->> OpenIDConnectPlugin: validate tokens
+    OpenIDConnectPlugin ->> OpenIDConnectPlugin: create internal tokens with granted permissions
+    OpenIDConnectPlugin -->> Web: set cookie refreshToken with <br>refresh token and redirect to<br> http://localhost:3000/?token=ZZZ&refreshToken=WWW&csrfToken=EEE
+`}
+/>
+
+#### External refresh
+<Chart
+  definition={`
+  sequenceDiagram
+    Web ->>+ Saleor: mutatuion ExternalRefresh with<br> refreshToken and csrf token as an input*
+    Saleor ->>+ OpenIDConnectPlugin: call <br>PluginManager.external_refresh
+    OpenIDConnectPlugin ->>OpenIDConnectPlugin: validate csrf and <br>refresh token structure
+    OpenIDConnectPlugin ->>+ OAuthProvider: refresh tokens
+    OAuthProvider -->>- OpenIDConnectPlugin: new tokens
+    OpenIDConnectPlugin ->> OpenIDConnectPlugin: create new internal tokens
+    OpenIDConnectPlugin -->>- Saleor: new tokens
+    Saleor ->>- Web: return access_token<br> and refresh_token
+`}
+/>
+
+> \* refresh token can be also passed as a cookie - refreshToken
+
+#### External verify
+<Chart
+  definition={`
+  sequenceDiagram
+    Web ->>+ Saleor: mutatuion ExternalVerify with<br> token as an input
+    Saleor ->>+ OpenIDConnectPlugin: call <br>PluginManager.external_verify
+    OpenIDConnectPlugin ->>OpenIDConnectPlugin: decode token
+    OpenIDConnectPlugin -->>- Saleor: return assigned user<br> and payload data
+    Saleor ->>- Web: return token data
+`}
+/>
+
+#### External logout
+<Chart
+  definition={`
+  sequenceDiagram
+    Web ->>+ Saleor: mutatuion ExternalLogout with<br> data to attach as an input
+    Saleor ->>+ OpenIDConnectPlugin: call <br>PluginManager.external_logout
+    OpenIDConnectPlugin ->>OpenIDConnectPlugin: prepare logout url:<br> logoutUrl + input data as GET params
+    OpenIDConnectPlugin -->>- Saleor: return logoutData
+    Saleor ->>- Web: return logoutData
+`}
+/>
+
+#### Authenticate user
+<Chart
+  definition={`
+  sequenceDiagram
+    Web ->>+ Saleor: Request to protected resources with token
+    Saleor ->>+ OpenIDConnectPlugin: Call in middleware <br>Plugin.authenticate_user
+    OpenIDConnectPlugin ->> OpenIDConnectPlugin: Decode and validate token
+    OpenIDConnectPlugin ->> OpenIDConnectPlugin: get user
+    OpenIDConnectPlugin ->> OpenIDConnectPlugin: assign permissions <br>from token
+    OpenIDConnectPlugin ->>- Saleor: User authenticated
+    Saleor ->>- Web: requested resources
+`}
+/>
+
+### OAuth permissions
+Saleor can use OAuth permissions assigned to a user. It maps internal permissions to OAuth scopes and attaches 
+it to the authorization request.
+
+```python
+def _get_oauth_session(self):
+    scope = "openid profile email "
+    scope += " ".join([f"saleor:{perm}" for perm in get_permissions_codename()])
```

Comment:
I'd rather see a full list of requested scopes than source code here. If we introduce new permissions we will be able to say things like "new since 3.0".